bin/bootstrax

```diff
@@ -474,31 +435,45 @@ def override_target(rd):
         return False
 
 
-def set_state(state):
+def set_state(state, update_fields=None):
     """Inform the bootstrax collection we're in a different state
 
     if state is None, leave state unchanged, just update heartbeat time
     """
     ping_dbs()
 
-    if state_doc_id is None:
-        log.debug("Not updating bootstrax doc: none selected. "
-                  "You are probably running a rogue bootstrax operation...")
-        return
-    update = {'time': now()}
-    if state is not None:
-        update['state'] = state
-    bs_coll.update_one(
-        {'_id': state_doc_id},
-        {'$set': update})
-
+    if not state:
+        # Find the last message of this host
+        previous_entry = bs_coll.find_one({'host': hostname,
+                                           'time': {'$lt': now(-timeouts['bootstrax_presumed_dead'])}},
+                                          sort=[('time', pymongo.DESCENDING)])
```

Comment:
I've had issues with `find_one` and the `sort` argument, I got around it via `find(query).sort().limit(1)`

Reply:
Thanks, I have tested this syntax and it did seem to work. I can double check.

Reply:
I tested this call on the runs db: `client.xenonnt.runs.find_one({'detector' : 'tpc'}, sort=[('time', -1)])['number']` and got `0`, which is obviously not correct.

Reply:
The reason why `client.xenonnt.runs.find_one({'detector' : 'tpc'}, sort=[('time', -1)])['number']` does not work is that there is no 'time' field in the rundoc. You can however sort by `_id` and that is what your original suggestion is doing when you:
`find(query).sort('_id',-1).limit(1)`

Reply:
We actually don't have any indexes in the rundocs. We might want to think about that at some point, it'll help performance in a few years once there's many more thousands of runs.